neuraxle/pipeline.py

```diff
@@ -202,52 +336,127 @@ def _transform_core(self, data_container: DataContainer) -> DataContainer:
+    def _load_checkpoint(self, data_container: DataContainer) -> Tuple[NamedTupleList, DataContainer]:
+        """
+        Try loading a pipeline cache with the passed data container.
+        If pipeline cache loading succeeds, find steps left to do,
+        and load the latest data container.
+
+        :param data_container: the data container to resume
+        :return: tuple(steps left to do, last checkpoint data container)
+        """
+        return self.steps_as_tuple, data_container
+
+
+class ResumablePipeline(Pipeline, ResumableStepMixin):
+    """
+    Fits and transform steps after latest checkpoint
+    """
+
+    def __init__(self, steps: NamedTupleList, pipeline_saver: PipelineSaver = None):
+        Pipeline.__init__(self, steps=steps)
+
+        if pipeline_saver is None:
+            self.pipeline_saver = JoblibPipelineSaver(DEFAULT_CACHE_FOLDER)
+        else:
+            self.pipeline_saver = pipeline_saver
+
+    def save(self, data_container: DataContainer):
         """
-        Find the steps left to do, and load the latest checkpoint step data container
+        Save the fitted parent pipeline with the passed data container
+
+        :param data_container: data container to save pipeline with
+        :return:
+        """
+        if self.parent_step is None:
+            self.pipeline_saver.save(self, data_container)
+        else:
+            self.parent_step.save(data_container)
+
+    def _load_checkpoint(self, data_container: DataContainer) -> Tuple[NamedTupleList, DataContainer]:
+        """
+        Try loading a pipeline cache with the passed data container.
+        If pipeline cache loading succeeds, find steps left to do,
+        and load the latest data container.
 
         :param data_container: the data container to resume
         :return: tuple(steps left to do, last checkpoint data container)
         """
-        new_starting_step_index = self._find_starting_step_index(data_container)
+        new_starting_step_index, starting_step_data_container = \
+            self._get_starting_step_info(data_container)
 
-        step = self.steps_as_tuple[new_starting_step_index]
+        if self.parent_step is None:
+            loaded_saved_pipeline = self._load_saved_pipeline(
+                starting_step_data_container,
+                new_starting_step_index
+            )
+
+            if not loaded_saved_pipeline:
+                return self.steps_as_tuple, data_container
+
+        step = self[new_starting_step_index]
         if isinstance(step, BaseCheckpointStep):
-            data_container = step.read_checkpoint(data_container)
+            starting_step_data_container = step.read_checkpoint(starting_step_data_container)
+
+        return self.steps_as_tuple[new_starting_step_index:], starting_step_data_container
+
+    def _load_saved_pipeline(
+            self,
+            starting_step_data_container: DataContainer,
```

Comment:
This method has many flaws. I already commented, it wasn't fixed.

Reply:
(contains many side effects, consider returning something other than a boolean).

Reply:
Re: I already commented on this in an old PR long ago.